src/public/js/rs.js

```diff
@@ -5,20 +5,50 @@ var RS = (function() {
 
   function encode(msg) {
     // dummy cipher for now -- advances each letter in message by ROT
-    return shift(msg, ROT);
+    return UTIL.shift(msg, ROT);
   }
 
   function decode(msg) {
     // dummy cipher for now -- advances each letter in message by -ROT
-    return shift(msg, -ROT);
+    return UTIL.shift(msg, -ROT);
   }
 
-  function shift(msg, rot) {
-    return msg.split('').reduce(function(accumulator, letter, index) {
-      var shiftedCharCode = letter.charCodeAt() + rot;
-      var shiftedLetter = String.fromCharCode(shiftedCharCode);
-      return accumulator + shiftedLetter;
-    }, '');
+  // Encodes using systematic BCH Reed-Solomon on Galois Field 256
+  // Message should be passed in as length k block
+  // k specifies message length
+  // n specifies block length (number of symbols transmitted)
+  // n - k is the number of code symbols added.
+  // Encoding procedure has the largest order coefficient on the left
+  // and the smallest on the right. IE: [1,2,3] -> x^2 + 2x + 3
+  // The largest order coeeficient is the [0] element of msg so as
+  // to maintain the ordering of [msg, code symbols] in the coded msg.
+  function encodeRSBlock(msg, n, k) {
+    var remainder = UTIL.polynomialDiv(msg, genPoly, n, k);
+    var codedMsg = new Uint8Array(n);
+    for (i = 0; i < k; i++) {
+      codedMsg[i] = msg[i];
+    }
+    for (i = k; i < n; i++) {
+      codedMsg[i] = remainder[i - k];
+    }
+    return codedMsg;
+  }
+
+  function decodeRSBlock(codedMsg, n, k) {
+    var remainder = UTIL.polynomialDiv(codedMsg, genPoly, n, k);
+    var errors = false;
+    for (i = 0; i < n - k; i++) {
```

Comment:
ohhh, also, I'm sorry I missed this earlier -- should be `for (var i = 0; ...)` otherwise i will end up on window instead of in the local scope and I'd worry that it might conflict with other i's if things were happening asynchronously (which I don't think is the case for us, but still good to avoid)

Surprised that the linter doesn't catch that one -- I'd think it would